### PR TITLE
rules: add mtd/by-name symlinks

### PR DESCRIPTION
**Pull request by tq-schiffern**
Add persistent symlinks for MTD devices like SPI-NOR flash, based on the partition names specified on the cmdline, in a Device Tree, or by other MTD partitioning parser drivers. Using the persistent name can be preferable to using the numbered /dev/mtdX device, as the latter can change depending on probe order or when partitioning has changed.

---

Open questions:
- Should we do something about /dev/mtdX-ro (read-only) devices? They don't have any MTD-specific udev attributes like `name`, so they at least don't interfere with the rule this PR adds, but it might be nice to have symlinks with an `-ro` suffix for them if there is a good way to achieve this
- MTD devices currently don't have `ENV{ID_PATH}`, so no `by-path` symlinks are added. Is this something we want?

### CODE REVIEW

File: rules.d/60-persistent-storage-mtd.rules [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# do not edit this file, it will be overwritten on update
+
+# persistent storage links: /dev/mtd/by-name
+
+ACTION=="remove", GOTO="persistent_storage_mtd_end"
+ENV{UDEV_DISABLE_PERSISTENT_STORAGE_MTD_FLAG}=="1", GOTO="persistent_storage_mtd_end"
```
Comment (yuwata) [inline rules.d/60-persistent-storage-mtd.rules]: I do not think this is necessary, at least now.

File: rules.d/60-persistent-storage-mtd.rules [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+SUBSYSTEM=="mtd", KERNEL=="mtd[0-9]*", ATTR{name}=="?*", SYMLINK+="mtd/by-name/$attr{name}"
```
Comment (yuwata) [inline rules.d/60-persistent-storage-mtd.rules]: Not a matter, but this file is for mtd devices, so, I'd like to place the condition `SUBSYSTEM=="mtd"` at the beginning of the file:
```
ACTION=="remove", GOTO="persistent_storage_mtd_end"
SUBSYSTEM!="mtd", GOTO="persistent_storage_mtd_end"
KERNEL!="mtd[0-9]*", GOTO="persistent_storage_mtd_end"

ATTR{name}=="?*", ...
```

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by yuwata** [comment]
> Open questions:
> 
>     * Should we do something about /dev/mtdX-ro (read-only) devices? They don't have any MTD-specific udev attributes like `name`, so they at least don't interfere with the rule this PR adds, but it might be nice to have symlinks with an `-ro` suffix for them if there is a good way to achieve this
> 
>     * MTD devices currently don't have `ENV{ID_PATH}`, so no `by-path` symlinks are added. Is this something we want?

Let's do that later if someone request that.

File: rules.d/60-persistent-storage-mtd.rules (hunk 1 above)
Comment (tq-schiffern) [reply rules.d/60-persistent-storage-mtd.rules]: Okay, removed.

File: rules.d/60-persistent-storage-mtd.rules (hunk 2 above)
Comment (tq-schiffern) [reply rules.d/60-persistent-storage-mtd.rules]: Done. I've also added another rule to explicitly skip `mtd*ro` devices, so they don't interfere if they ever get a `name` attr in the future.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**